src/main/java/com/umbr3114/controllers/CommentController.java

```diff
@@ -0,0 +1,160 @@
+            }
+            commentModelList = getByDate(postId, date);
+        }else{
+            commentModelList = getAfterId(postId, startAfterId);
+        }
+        if(postId == null){
+            return "not ok";
+        }
+        CommentListingModel commentList = new CommentListingModel();
+        commentList.count = commentModelList.size();
+        if (commentModelList.size() > 0){
+            commentList.lastId = commentModelList.get(commentModelList.size()-1)._id.toString();
+            commentList.postId = commentModelList.get(0).getPostIdString();
+        }
+        commentList.comments = commentModelList;
+        return commentList;
+
+    });
+
+    private static List<CommentModel> getAfterId(String postId, String startAfterId){
+        JacksonMongoCollection<CommentModel> mongoCollectionComments =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        JacksonMongoCollection<PostModel> mongoCollectionPost =
+                new CollectionFactory<PostModel>(ServiceLocator.getService().dbService(), PostModel.class).getCollection();
+        FindIterable<CommentModel> iterable;
+        PostModel post = mongoCollectionPost.findOne(Filters.eq("_id", postId));
+        List<CommentModel> commentModels = new ArrayList<>();
+        if(post == null){
+            return commentModels;
+        }else{
+            iterable = mongoCollectionComments.find(Filters.and(Filters.eq("post_id", postId), Filters.gt("_id", startAfterId.toString())));
+        }
+        iterable.limit(10);
+        iterable.into(commentModels);
+        return commentModels;
+    }
+
+    private static List<CommentModel> getByDate(String postId, String date){
+        JacksonMongoCollection<CommentModel> mongoCollectionComments =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        FindIterable<CommentModel> iterable;
+        List<CommentModel> commentList = new ArrayList<>();
+        iterable = mongoCollectionComments.find(Filters.and(eq("dateString", date), Filters.eq("post_id", postId))).limit(10);
+        iterable.into(commentList);
+        return commentList;
+    }
+
+    public static Route updateComments = ((request, response) -> {
+        JacksonMongoCollection<CommentModel> mongoCollection =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        SessionManager sessionManager = new SparkSessionManager(request);
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        CommentModel commentModel;
+        //PermissionChecker permissionChecker;
+        String commentId = paramHelper.valueOf("_id");
+        String bodyText = paramHelper.valueOf("bodyText");
+        commentModel = mongoCollection.findOne(eq("_id", new ObjectId(commentId)));
```

Comment:
commentId is not null checked, this might be why the server crashes with a 500 error